sycl/doc/extensions/SPIRV/SPV_INTEL_fpga_buffer_location.asciidoc

```diff
@@ -0,0 +1,121 @@
+SPV_INTEL_fpga_buffer_location
+==============================
+
+== Name Strings
+
+SPV_INTEL_fpga_buffer_location
+
+== Contact
+
+To report problems with this extension, please open a new issue at:
+
+https://github.com/KhronosGroup/SPIRV-Headers
+
+== Contributors
+
+- Joe Garvey, Intel +
+
+== Notice
+
+Copyright (c) 2019-2020 Intel Corporation.  All rights reserved.
+
+== Status
+
+Draft
+
+== Version
+
+[width="40%",cols="25,25"]
+|========================================
+| Last Modified Date | {docdate}
+| Revision           | A
+|========================================
+
+== Dependencies
+
+This extension is written against the SPIR-V Specification,
+Version 1.5 Revision 2.
+
+This extension requires SPIR-V 1.0.
+
+== Overview
+
+This extension adds a function parameter decoration that is useful for FPGA targets.  This decoration indicates that a particular global memory pointer can only access a particular physical memory location.  Knowing this information at compile time can allow FPGA compilers to generate load store units of lower area for accesses done through such a pointer.  
+
+== Extension Name
+To use this extension within a SPIR-V module, the following *OpExtension* must be present in the module:
+
+----
+OpExtension "SPV_INTEL_fpga_buffer_location"
+----
+
+== New Capabilities
+This extension introduces a new capability:
+
+----
+FPGABufferLocationINTEL
+----
+
+== Token Number Assignments
+
+--
+[width="40%"]
+[cols="70%,30%"]
+[grid="rows"]
+|====
+|FPGABufferLocationINTEL |5920
+|BufferLocationINTEL     |5921
+|==== 
+--
+
+== Modifications to the SPIR-V Specification, Version 1.5
+
+=== Decoration
+
+Modify Section 3.20, Decoration, adding these rows to the Decoration table:
+
+--
+[options="header"]
+|====
+2+^| Decoration | Extra Operands | Enabling Capabilities
+| 5921 | *BufferLocationINTEL* + 
+Apply to a pointer *OpFunctionParameter*.  Indicates that the pointer may only point into the physical memory indentified by the subsequent literal number operand.  | Literal +
```

Comment:
```suggestion
Apply to a  *OpFunctionParameter*.  Indicates that the pointer may only point into the physical memory indentified by the subsequent literal number operand.  | Literal +
```

Reply:
@MrSidims after our discussion, is this still required?
In SPIR-V, I don't think we should be able to apply the decoration on a non-pointer `OpFunctionParameter`

Reply:
agreed